railties/lib/rails/app_updater.rb

```diff
@@ -29,6 +29,7 @@ def generator_options
           options[:skip_action_text]    = !defined?(ActionText::Engine)
           options[:skip_action_cable]   = !defined?(ActionCable::Engine)
           options[:skip_test]           = !defined?(Rails::TestUnitRailtie)
+          options[:skip_system_test]    = Rails.application.config.generators.system_tests.nil?
```

Comment:
For reference, `options[:skip_system_test]` triggers `!depends_on_system_test?`:

https://github.com/rails/rails/blob/88921001efc47764f44d225ef32aaadaaa489bcd/railties/lib/rails/generators/app_base.rb#L479-L481

which triggers `config.generators.system_tests = nil`:

https://github.com/rails/rails/blob/88921001efc47764f44d225ef32aaadaaa489bcd/railties/lib/rails/generators/rails/app/templates/config/application.rb.tt#L47-L51